### PR TITLE
Feature/35712 redesign of members list

### PR DESCRIPTION
**Pull request by burczu**
### Details

In this PR the list of members displayed on the Workspace Members Page is re-designed to match the new design.

The new design for reference:

Screenshot 2024-02-22 at 11 15 21

### Fixed Issues

$ https://github.com/Expensify/App/issues/35712
PROPOSAL: https://docs.google.com/document/d/1gk3xqOs7epMbUrSSiX8K7YcqfPLVgqEos0sf-D-GMDA/edit#heading=h.qjcht79r25s1

### Tests

1. LogIn to the App
2. Open any Workspace
3. Open Members page
4. Check if the members list looks the same as on the design.

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as above.

### QA Steps

Same as above.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

Screenshot 2024-02-22 at 12 47 29

</details>

<details>
<summary>Android: mWeb Chrome</summary>

Screenshot 2024-02-22 at 12 42 20

</details>

<details>
<summary>iOS: Native</summary>

Screenshot 2024-02-22 at 12 40 35

</details>

<details>
<summary>iOS: mWeb Safari</summary>

Screenshot 2024-02-22 at 12 41 34

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

Screenshot 2024-02-22 at 12 33 51

</details>

<details>
<summary>MacOS: Desktop</summary>

Screenshot 2024-02-22 at 12 35 07

</details>

### CODE REVIEW

File: src/components/SelectionList/TableListItem.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    onDismissError,
+    shouldPreventDefaultFocusOnSelectRow,
+    rightHandSideComponent,
+}: TableListItemProps) {
+    const styles = useThemeStyles();
+    const theme = useTheme();
+    const StyleUtils = useStyleUtils();
+
+    const focusedBackgroundColor = styles.sidebarLinkActive.backgroundColor;
+    const hoveredBackgroundColor = !!styles.sidebarLinkHover && 'backgroundColor' in styles.sidebarLinkHover ? styles.sidebarLinkHover.backgroundColor : theme.sidebar;
```
Comment (ArekChr) [inline src/components/SelectionList/TableListItem.tsx:27]:
```suggestion
    const hoveredBackgroundColor = !!styles.sidebarLinkHover?.backgroundColor ? styles.sidebarLinkHover.backgroundColor : theme.sidebar;
```

**Comment by ArekChr** [comment]
cc: @Expensify/design

**Comment by getusha** [comment]
@shawnborton i wonder if we should also use the new Badge design here?

**Comment by shawnborton** [comment]
Looking pretty good! 

Totally agree that we should use the new badge component. Has that been merged into main yet? 

For this area here:
CleanShot 2024-02-22 at 07 47 57@2x

I would think that the "Role" heading would be right-aligned since it's the last column. It also feels like each row has too much padding-right - I would expect the badge to be further over to the right. How much right-padding is currently being applied?

**Comment by shawnborton** [comment]
Ah okay, you can ignore me then. cc @trjExpensify @JmillsExpensify @Expensify/design for confirmation.

Can you show me how much right padding is being used though?

File: src/pages/workspace/WorkspaceMembersPage.js [hunk 2]
```diff
@@ -371,11 +352,7 @@ function WorkspaceMembersPage(props) {
                     !_.isEmpty(policyMember.errors),
                 text: props.formatPhoneNumber(PersonalDetailsUtils.getDisplayNameOrDefault(details)),
                 alternateText: props.formatPhoneNumber(details.login),
-                rightElement: isAdmin ? (
-                    <View style={[styles.badge, styles.peopleBadge]}>
-                        <Text style={styles.peopleBadgeText}>{props.translate('common.admin')}</Text>
-                    </View>
-                ) : null,
+                rightElement: roleBadge,
```
Comment (getusha) [inline src/pages/workspace/WorkspaceMembersPage.js:355]:
```suggestion
                    <Badge
                        text={isOwner ? props.translate('common.owner') : props.translate('common.admin')}
                        textStyles={styles.textStrong}
                        badgeStyles={[
                            styles.alignSelfCenter,
                            styles.badgeBordered,
                        ]}
                    />
```

we can use the new badge like this, we can also avoid creating a separate variable for it.

Comment (getusha) [reply src/pages/workspace/WorkspaceMembersPage.js:355]: https://github.com/Expensify/App/blob/29ad8787c7c889271c117dd4420109f9be8e180d/src/components/MenuItem.tsx#L574-L583

**Comment by burczu** [comment]
@shawnborton Alright, fixed the badge - thanks. It looks like below now:

Screenshot 2024-02-22 at 14 24 37

**Comment by shawnborton** [comment]
Lovely! How much vertical padding does each row have? I think we want to be consistent and give these rows 16 vPadding and 16 hPadding. It seems like maybe there isn't quite as much vertical padding from the screenshot you showed.

**Comment by burczu** [comment]
> Lovely! How much vertical padding does each row have? I think we want to be consistent and give these rows 16 vPadding and 16 hPadding. It seems like maybe there isn't quite as much vertical padding from the screenshot you showed.

It has 12px of vertical padding - taken from Figma. I can change it to 16px

**Comment by getusha** [comment]
BUG: removing a member offline leaves blank component.

https://github.com/Expensify/App/assets/75031127/a4aca654-08a3-4174-9d23-703f1c075ebf

**Comment by burczu** [comment]
> BUG: removing a member offline leaves blank component.
> 
>  Screen.Recording.2024-02-22.at.4.23.41.PM.mov

@getusha Thanks for pointing this out - I didn't touch the removal functionality yet. I'll be working on it in the next issue so I think I'll address this there (it's not a breaking issue I guess). What do you think?

**Comment by shawnborton** [comment]
Ah, got it. Maybe that is intended then, @Expensify/design thoughts on 12px vs 16px for vertical padding? Maybe @burczu can show us a comparison

**Comment by burczu** [comment]
> Ah, got it. Maybe that is intended then, @Expensify/design thoughts on 12px vs 16px for vertical padding? Maybe @burczu can show us a comparison

@shawnborton This is how it looks with 16px vertical padding (very minor distinction for me tbh):

Screenshot 2024-02-22 at 14 34 55

**Comment by shawnborton** [comment]
Okay personally I lean towards wanting 16px for both v and h. Let's see if Danny and Jon have any thoughts there too though.

File: src/styles/index.ts [hunk 3]
```diff
@@ -3116,15 +3116,17 @@ const styles = (theme: ThemeColors) =>
...
-            ...spacing.ph3,
-            ...spacing.ml3,
+            backgroundColor: theme.buttonDefaultBG,
+            paddingHorizontal: 6,
+            ...flex.justifyContentCenter,
+            ...borders.br1,
+            ...spacing.mh0,
         },
 
         peopleBadgeText: {
```
Comment (getusha) [inline src/styles/index.ts:3126]: can we remove this? looks like it's not needed anymore.

**Comment by getusha** [comment]
@burczu lets also update the Storybook & we've got a merge conflict.

**Comment by dannymcclain** [comment]
@shawnborton I'm down to standardize the padding for "tables" everywhere (here, money page, tags, categories, etc.). Your most recent "[tightened spacing](https://www.figma.com/file/l38UCEdwCZGejAh3BW6N7e/Money-Page?type=design&node-id=582-102076&mode=design&t=cfMKRPE2aqucQEbn-11)" for the money page had the padding at 12px—do we want to standardize on 12px or 16px? I'm good either way, I kinda just want them to be the same for all of these types of rows.

Also, this "Role" column label still looks misaligned to me (I realize it is technically properly aligned left, but I think it looks weird). I think we need to center align the column header like we're doing for the "Action" column on the money page designs.

CleanShot 2024-02-22 at 08 25 31@2x

**Comment by getusha** [comment]
>@getusha Thanks for pointing this out - I didn't touch the removal functionality yet. I'll be working on it in the next issue so I think I'll address this there (it's not a breaking issue I guess). What do you think?

Yes, it is minor, can be handled later.

**Comment by shawnborton** [comment]
@dannymcclain I think I am team 16px right now, will tag ya in Figma where I'm using that currently.

**Comment by luacmartins** [comment]
Thanks for the reviews so far! @burczu would you mind addressing the latest comments and fixing the conflict so we can get this merged today?

**Comment by burczu** [comment]
@getusha your comments addressed + storybook updated
@shawnborton 16px vertical padding applied
@dannymcclain the Role column label center aligned as requested
@luacmartins conflicts solved and all the comments and requests addressed, I believe

**Comment by getusha** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

File: src/languages/es.ts [hunk 4]
```diff
@@ -195,6 +195,7 @@ export default {
         iAcceptThe: 'Acepto los ',
         remove: 'Eliminar',
         admin: 'Administrador',
+        owner: 'Poseedor',
```
Comment (luacmartins) [inline src/languages/es.ts:198]: NAB I think this should be `Dueño`

File: src/pages/workspace/WorkspaceMembersPage.tsx [hunk 5]
```diff
@@ -383,23 +353,34 @@ function WorkspaceMembersPage({policyMembers, personalDetails, route, policy, se
...
-                // eslint-disable-next-line @typescript-eslint/naming-convention
-                messages={{'0': 'workspace.people.addedWithPrimary'}}
-                containerStyles={[styles.pb5, styles.ph5]}
-                onClose={() => Policy.dismissAddedWithPrimaryLoginMessages(policyID)}
-            />
-        );
-    };
+    const getHeaderContent = () => (
+        <>
+            <Text style={[styles.pl5, styles.mb5, styles.mt3]}>{translate('workspace.people.membersListTitle')}</Text>
```
Comment (luacmartins) [inline src/pages/workspace/WorkspaceMembersPage.tsx:361]: NAB I think this text should use `textSupporting` color

**Review by luacmartins — APPROVED** [review]
There are a couple of small issues, but I think we can address those in a follow up since this PR unblocks Categories main page and the whole dependency chain of issues there.

**Review by getusha — APPROVED** [review]
Looks good to me!

**Review by luacmartins — APPROVED** [review]

File: src/pages/workspace/WorkspaceMembersPage.tsx [hunk 6]
```diff
@@ -339,8 +301,20 @@ function WorkspaceMembersPage({policyMembers, personalDetails, route, policy, se
                 }
             }
 
+            const isOwner = policy?.owner === details.login;
             const isAdmin = session?.email === details.login || policyMember.role === CONST.POLICY.ROLE.ADMIN;
 
+            let roleBadge = null;
+            if (isOwner || isAdmin) {
+                roleBadge = (
```
Comment (rlinoz) [inline src/pages/workspace/WorkspaceMembersPage.tsx:309]: The badge wasn't being styled properly when removing admin while offline in the native apps: https://github.com/Expensify/App/issues/37776

**Pull request merged**